plugin/github/env.go

```diff
@@ -100,15 +100,30 @@ func parseOwner(s string) (owner string) {
 }
 
 func getWith(envVars map[string]string) (map[string]string, error) {
-	if val, ok := envVars["PLUGIN_WITH"]; ok {
-		with, err := strToMap(val)
-		if err != nil {
-			return nil, errors.Wrap(err, "with attribute is not of map type with key & value as string")
+	_, exists := envVars["PLUGIN_WITH"]
+
+	// TODO: Remove this once corresponding CI Manager changes are released on prod envs
+	if exists {
+		// PLUGIN_WITH is being set in older flow
+		if val, ok := envVars["PLUGIN_WITH"]; ok {
+			with, err := strToMap(val)
+			if err != nil {
+				return nil, errors.Wrap(err, "with attribute is not of map type with key & value as string")
+			}
+			return with, nil
 		}
-
-		return with, nil
+		return nil, nil
+	} else {
+		m := make(map[string]string)
+		prefix := "PLUGIN_WITH_"
```

Comment:
keep `PLUGIN_WITH` in a single place at top of the function. And prefix check could be just for `PLUGIN_WITH`. You could reuse the string everywhere in this function.
`pluginWithVar := "PLUGIN_WITH"`

Reply:
I'm passing PLUGIN_WITH_<varname> from ci manager, so we should be checking for prefix PLUGIN_WITH_